app/services/waste_carriers_engine/discern_uk_number_type_service.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module WasteCarriersEngine
+  class DiscernUkNumberTypeService < BaseService
+    def run(phone_number:)
+      # Remove any spaces in the phone number
+      phone_number = phone_number.gsub(" ", "")
+
+      # UK mobile numbers start with 07 or +447
+      return "mobile" if phone_number =~ /^(07|\+447)/
+
+      # UK landline numbers start with 01 or 02, or +441 or +442
```

Comment:
It's a bit more complicated. 070 and 076 are not mobile. 03x and some 08x, 09x are non-geographic. 055 are corporate.
https://www.ofcom.org.uk/__data/assets/pdf_file/0013/102613/national-numbering-plan.pdf